commons/push/integration/garmin/garmin_push_activity_details.avsc

```diff
@@ -0,0 +1,46 @@
+        {"name": "numberOfActiveLengths", "type": ["null", "int"], "doc": "Number of Active Lengths.", "default": null},
+        {"name": "averageHeartRateInBeatsPerMinute", "type": ["null", "int"], "doc": "Average of heart rate values captured during the last 7 days, in beats per minute. The average heart rate value for the monitoring period can be calculated based on the data from timeOffsetHeartRateSamples.", "default": null},
+        {"name": "maxHeartRateInBeatsPerMinute", "type": ["null", "int"], "doc": "Maximum of heart rate values captured during the monitoring period, in beats per minute.", "default": null},
+        {"name": "startingLatitudeInDegree", "type": ["null", "double"], "doc": "Starting Latitude in Degree", "default": null},
+        {"name": "startingLongitudeInDegree", "type": ["null", "double"], "doc": "Starting Longitude in Degree", "default": null},
+        {"name": "steps", "type": ["null", "int"], "doc": "Count of steps recorded during the monitoring period.", "default": null},
+        {"name": "totalElevationGainInMeters", "type": ["null", "double"], "doc": "Total Elevation Gain in meters", "default": null},
+        {"name": "totalElevationLossInMeters", "type": ["null", "double"], "doc": "Total Elevation Loss in meters", "default": null},
+        {"name": "isParent", "type": ["null", "boolean"], "doc": "If present and set to true, this activity is the parent activity of one or more child activities that should also be made available in the data feed to the partner. An activity of type MULTI_SPORT is an example of a parent activity.", "default": null},
+        {"name": "parentSummaryId", "type": ["null", "string"], "doc": "If present, this is the summaryId of the related parent activity. An activity of type CYCLING with a parent activity of type MULTI_SPORT is an example of this type of relationship.", "default": null},
+        {"name": "manual", "type": ["null", "boolean"], "doc": "Indicates that the activity was manually entered directly on the Connect site. This property will only exist for manual activities.", "default": null},
+        {"name": "deviceName", "type": ["null", "string"], "doc": "Only Fitness Activities are associated with a specific Garmin device rather than the user’s overall account. If a user wears two devices at once during the same time and starts a Fitness Activity on each then both will generate separate Activity summaries with two different deviceNames. Always ‘unknown’ for manually created activities.", "default": null}
+      ]
+      }
+    },
+    {"name": "samples", "type": { "type": "array", "items" : "GarminActivityDetailsSample", "default": [] }, "doc": "Each activity detail contains an optional list of samples. The samples list will be empty if the activity is manual or details are not supported by the device. Samples may be as frequent as once per second, and values should be considered valid until the next sample."}
```

Comment:
Each of the samples should get their own record. You can add the summaryId to the samples to have a reference